### PR TITLE
Update azure backend storage sdk 

### PR DESCRIPTION
**Pull request by mbfrahry**

### CODE REVIEW

File: backend/remote-state/azure/backend_state.go [hunk 1]
```diff
@@ -21,23 +22,22 @@ const (
...
-	container := client.GetContainerReference(b.containerName)
-	resp, err := container.ListBlobs(params)
+	resp, err := client.ListBlobs(ctx, b.armClient.storageAccountName, b.containerName, params)
 	if err != nil {
 		return nil, err
 	}
 
 	envs := map[string]struct{}{}
-	for _, obj := range resp.Blobs {
+	for _, obj := range resp.Blobs.Blobs {
```
[the remark this answers is not in the document]
Comment (mbfrahry) [reply backend/remote-state/azure/backend_state.go:40]: Hmmm, I'm not seeing a way blobs could be nil from the [response](https://github.com/tombuildsstuff/giovanni/blob/master/storage/2018-11-09/blob/containers/list_blobs.go#L30)

File: backend/remote-state/azure/client.go [hunk 2]
```diff
@@ -143,41 +125,44 @@ func (c *RemoteClient) Lock(info *state.LockInfo) (string, error) {
...
-		if v := stateMgr.State(); v == nil {
-			if err := stateMgr.WriteState(states.NewState()); err != nil {
-				return "", fmt.Errorf("Failed to write empty state for locking: %s", err)
-			}
-			if err := stateMgr.PersistState(); err != nil {
-				return "", fmt.Errorf("Failed to persist empty state for locking: %s", err)
-			}
+		contentType := "application/json"
+		putGOptions := blobs.PutBlockBlobInput{
+			ContentType: &contentType,
```
[the remark this answers is not in the document]
Comment (mbfrahry) [reply backend/remote-state/azure/client.go:145]: I'm not sure. This is the only place we acquire a lease in the backend

[the remark this answers is not in the document]
Comment (tombuildsstuff) [reply backend/remote-state/azure/client.go:145]: actually checking [the API docs](https://docs.microsoft.com/en-us/rest/api/storageservices/put-blob) this needs to be two separate steps, so this is fine:

> Required if the blob has an active lease. To perform this operation on a blob with an active lease, specify the valid lease ID for this header.

**Comment by tombuildsstuff** [comment]
Tests pass:

```
$ TF_AZURE_TEST=1 envchain azurerm go test -v ./backend/remote-state/azure/ -timeout=60m
=== RUN   TestBackend_impl
--- PASS: TestBackend_impl (0.00s)
=== RUN   TestBackendConfig
--- PASS: TestBackendConfig (0.00s)
=== RUN   TestBackendAccessKeyBasic
--- PASS: TestBackendAccessKeyBasic (130.21s)
=== RUN   TestBackendManagedServiceIdentityBasic
--- SKIP: TestBackendManagedServiceIdentityBasic (0.00s)
=== RUN   TestBackendSASTokenBasic
--- PASS: TestBackendSASTokenBasic (114.67s)
=== RUN   TestBackendServicePrincipalBasic
--- PASS: TestBackendServicePrincipalBasic (131.59s)
=== RUN   TestBackendServicePrincipalCustomEndpoint
--- SKIP: TestBackendServicePrincipalCustomEndpoint (0.00s)
=== RUN   TestBackendAccessKeyLocked
--- PASS: TestBackendAccessKeyLocked (128.45s)
=== RUN   TestBackendServicePrincipalLocked
--- PASS: TestBackendServicePrincipalLocked (128.32s)
=== RUN   TestRemoteClient_impl
--- PASS: TestRemoteClient_impl (0.00s)
=== RUN   TestRemoteClientAccessKeyBasic
--- PASS: TestRemoteClientAccessKeyBasic (111.05s)
=== RUN   TestRemoteClientManagedServiceIdentityBasic
--- SKIP: TestRemoteClientManagedServiceIdentityBasic (0.00s)
=== RUN   TestRemoteClientSasTokenBasic
--- PASS: TestRemoteClientSasTokenBasic (126.44s)
=== RUN   TestRemoteClientServicePrincipalBasic
--- PASS: TestRemoteClientServicePrincipalBasic (112.29s)
=== RUN   TestRemoteClientAccessKeyLocks
--- PASS: TestRemoteClientAccessKeyLocks (126.90s)
=== RUN   TestRemoteClientServicePrincipalLocks
--- PASS: TestRemoteClientServicePrincipalLocks (128.15s)
=== RUN   TestPutMaintainsMetaData
--- PASS: TestPutMaintainsMetaData (126.71s)
PASS
ok  	github.com/hashicorp/terraform/backend/remote-state/azure	1364.877s
```

**Pull request merged**